### ISSUE TITLE
Expand Tracker Search

### ISSUE THREAD

**Issue by nopeha** [mihonapp/mihon#514]
### Describe your suggested feature

There are times when a tracker I know has a manga, but it doesn't appear in the list of available manga when I search the tracker through mihon. I sometimes find I have to come back to an entry in my library and hope that the next time I search a tracker it pops up so I can select it.

What I'm suggesting is a "load more" feature to allow users to search a bit further. I wouldn't suggest loading up all search results at once (seems bad for both mihon and tracking service), but having the ability to search more results would be helpful.

If this is optional for users, I think that would best to keep them from spamming trackers. If there is a toggle in the settings to expand tracking services searches with a warning regarding spamming the tracker could result in consequences from the tracking sevice (if that's an issue), I believe it'd be best.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.4](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by MajorTanya** [comment]
You could try variants of the title, like Romaji, Japanese, or English titles when you search a tracker.

Most of them try to return results ordered by relevance, so a "Show More" would just show even less relevant entries.

**Comment by BrutuZ** [comment]
I am pretty sure you can search for the ID in these cases

**Comment by MajorTanya** [comment]
> > I am pretty sure you can search for the ID in these cases
> 
> Tried that for the title with MAL, didn't see it come up (https://myanimelist.net/manga/156200). It did, however, prioritize showing me results with "200" in the title when searching for 156200. If I need to include something else for the search for ID, please let me know. If it works for certain trackers that may be why I didn't see it for MAL, in this instance.
> 
> Even if the ID method did work, that isn't very intuitive or directed by the app to do for the average user who may not know there is even an entry on a tracker. I only know there is one on MAL because a source has it listed. This isn't always the case. If it did work by ID, I'd suggest the placeholder text of "Search..." should change to "Search by Work Name or ID" or something closer. That'd be a separate ticket, though.

ID search uses the id: prefix, as seen here
![Screenshot_2024-03-14-20-35-43-995_app.mihonya.png](https://github.com/mihonapp/mihon/assets/39014446/3c82bcb7-77c9-484e-aca5-0d3c3fdc015a)

**Comment by BrutuZ** [comment]
For MAL you have to prefix the ID with `id:`, so it would be `id:156200`

**Comment by BrutuZ** [comment]
> "check alternative titles" when the source provides them

Non-standard, so not happening. Can only wish for such a field on `extensions-lib` v2

**Comment by BrutuZ** [comment]
> Perhaps scan the description for a format ?

Again, non-standard. Each source is free to slap whatever they want and each extension can process it on their own way or even ignore the information completely.
Are the alternatives provided separately on the site/API or built into the description? If the latter, does the "list" have a header? What's the separator, comma, semi-colon, pipe, newlines, newlines with prefixes...

Too many variables to account for. It needs to be a proper field on the [SManga](https://github.com/mihonapp/mihon/blob/main/source-api/src/commonMain/kotlin/eu/kanade/tachiyomi/source/model/SManga.kt) interface so that extensions handle processing the information when available and declare it consistently.
